### PR TITLE
Don't create AsyncLoader instance at import time.

### PR DESCRIPTION
**Pull request by mdickinson**
Currently, `import mapping.enable.api` creates a new `AsyncLoader` instance and starts a Python thread. This PR makes that creation lazy, waiting until the `AsyncLoader` is needed by the `HTTPTileManager` before creating it.

### CODE REVIEW

File: mapping/enable/async_loader.py [hunk 1]
```diff
@@ -2,7 +2,8 @@
 import asyncore
 import Queue
 import socket
-from threading import Thread
+from threading import Event, Thread
+from time import time as _time
```
Comment (corranwebster) [inline mapping/enable/async_loader.py:6]: Stray import?

Comment (corranwebster) [reply mapping/enable/async_loader.py:6]: No - apparently fixing a bug in code that was elided (either that, or github isn't showing all changes).

File: mapping/enable/async_loader.py [hunk 2]
```diff
@@ -12,33 +13,43 @@ class AsyncLoader(HasTraits):
...
+        self._stop_signal.set()
+        self._thread.join()
+
     def put(self, request):
         self._queue.put(request)
 
     ### Private interface ##################################################
     
     _thread = Any
+    _stop_signal = Any
```
Comment (corranwebster) [inline mapping/enable/async_loader.py:26]: Should this be `Instance(Event)` (and I guess similarly with `_thread` being `Instance(Thread)`) or is there a reason we want to be more general than this?

In fact, `Instance(Event, ())` would mean that the default handler would not be needed.

Comment (mdickinson) [reply mapping/enable/async_loader.py:26]: `Instance(Event)` unfortunately isn't right: `threading.Event` is a factory function, not a class.

File: mapping/enable/async_loader.py (hunk 1 above)
Comment (mdickinson) [reply mapping/enable/async_loader.py:6]: Yes, the non-None timeout branch of `get_all` was broken.

**Comment by senganal** [comment]
@mdickinson The PR needed a master merge.. This LGTM.. Can i merge this PR ?

**Comment by mdickinson** [comment]
@senganal: Sure, go ahead. The PR doesn't go as far as I would like, but it might reasonably be considered an improvement over the current situation.

**Pull request merged**